aggregator/aggregator.go

```diff
@@ -157,24 +197,48 @@ func (a *Aggregator) AddOrCreate(key string, ts uint32, quantized uint, value fl
 
 // Flush finalizes and removes aggregations that are due
 func (a *Aggregator) Flush(ts uint) {
+	flushWaiting.Inc(1)
 	flushes.Add()
-	for k, proc := range a.aggregations {
-		if k.ts <= ts {
+	flushWaiting.Dec(1)
+	defer flushes.Done()
+
+	pos := -1 // will track the pos of the last ts position that was successfully processed
+	for i, quantized := range a.tsList {
+		if quantized > ts {
```

Comment:
if it's `>` instead of `>=` this makes the `ts` inclusive at the end. but when the `ts` is calculated on line `:343` (`now.Add(-time.Duration(a.Wait) * time.Second)`) it only subtracts the wait time from the current second, so shouldn't that condition be `>=`? Otherwise if we just passed the second border then this condition implements the wait time to only be at least `ts-1`.

Reply:
This description a bit hard to follow, but let's turn this into an example which i think demonstrates the problem you're describing, but in a clearer way.
"if we just passed the second border", so for example now = 15:00:42,001 (3 PM, 42 sec and 1 ms)
wait = 22s
this makes thresh = 15:00:20,001, after `.Unix()`, Flush is called with a timestamp of 15:00:20). 
imagine we have a quantized value of 15:00:20, `quantized > ts` is false so we do not break and process/flush this aggregate.
If i'm reading your argument right, you're saying this value should not be flushed because one more second should have passed for it to be flushed.
But from what I can see, 22 seconds and 1 ms have passed, so the wait period has passed so we should flush.

Reply:
Also FYI this was very recently changed. see #279 for more information